src/audio-manager.ts

```diff
@@ -331,41 +373,81 @@ export class AudioManager {
      *
      * @warning This does not work for one-shots!
      *
-     * @param sourceId Identifier of the audio source that should be stopped.
-     * @param playId Optional parameter that specifies the exact audio that should be stopped.
-     * If not provided, all playing instances of the given sourceId will be stopped.
+     * @param playId Specifies the exact audio that should be stopped.
+     * @param sourceId Optional paramter, left for backwards compatablity. Doesnt do anything
      *
      * @note Obtain the playId from the play() method.
      * @see play
+     */ // @todo jonathan: Now it works like it always should have. Is this fine for backwards compat?
+    stop(playId: number, sourceId?: number) {
+        this._playerCache.forEach((player) => {
+            if (player.callerID === playId) {
+                player.stop();
+                return;
+            }
+        });
+    }
+
+    /**
+     * Pauses a playing audio.
+     *
+     * @param playId Id of the source that should be paused.
      */
-    stop(sourceId: number, playId?: number) {
-        if (playId) {
-            this._busyPlayers.get(playId)?.stopAndFree();
-            return;
-        }
-        this._busyPlayers.forEach((player) => {
-            if (player.bufferId >> SHIFT_AMOUNT === sourceId) {
-                player.stopAndFree();
+    pause(playId: number) {
+        this._playerCache.forEach((player) => {
+            if (player.callerID === playId) {
+                player.pause();
+                return;
+            }
+        });
+    }
+
+    /**
+     * Resumes a paused audio.
+     *
+     * @param playId Id of the source that should be resumed.
+     */
+    resume(playId: number) {
+        this._playerCache.forEach((player) => {
+            if (player.callerID === playId) {
+                player.resume();
+                return;
             }
         });
     }
 
     /**
      * Stops playback of all one-shot players.
+     * @deprecated since >1.2.0, use stop() instead.
      */
     stopOneShots() {
-        for (const node of this._oneShotCache) {
-            node.stop();
-        }
+        // @todo jonathan: How is this handled the best? Log a Warning?
```

Comment:
@Squareys also not sure what to do here. But i guess nobody was using it anyway...

Reply:
The problem is that one-shots and the other palyers are no longer separated, so you cannot pause one or the other, right?

As in the `stopAll()` is still a thing? Assuming that nobody was using it is always a bit risky 😌 

I'm not sure why you joined the caches anyway, was that necessary for the pausing? :D 
You could have an array of player types, which keeps track of whether a player is one-shot or not to still be able to provide this functionality.

Reply:
`stopAll()` is still a thing, yes. I know its risky :worried: 

I removed them because both player types were getting very similiar and i found a much better way of managing the players as to having them seperate (with seperate amounts of players which was inefficeient management). Also the regular players used a HashMap to  manage 16 player :rofl: and a list to manage the free players.
Its just way nicer now and there is no need to have to types of players.